sdk/nodejs/appservice/plan.ts

```diff
@@ -235,7 +237,7 @@ export interface PlanState {
     /**
      * The kind of the App Service Plan to create. Possible values are `Windows` (also available as `App`), `Linux`, `elastic` (for Premium Consumption) and `FunctionApp` (for a Consumption Plan). Defaults to `Windows`. Changing this forces a new resource to be created.
      */
-    readonly kind?: pulumi.Input<string>;
+    readonly kind?: pulumi.Input<string | Kind>;
```

Comment:
I would actually use `pulumi.Input<Kind>` here. It's as close to a non-breaking change as we can get with such refactorings...

Reply:
This is already the case. It's a pulumi.Input of string OR kind

Reply:
I mean "only `Kind`", restricting the values. It's only breaking if people used some funky casing, and @jen20 did similar restrictions in AWS

Reply:
Oh you don't think we should accept both? That feels like it is a breaking change - the casing is actually insensitive in the underlying provider